### PR TITLE
Add "Fixing TDVT Tests" Page, Update tdvt page

### PR DESCRIPTION
**Pull request by pvanderknyff**
Part of the TDVT documentation upgrade. Adds a new page going more in-depth on fixing a single TDVT test case.

This PR does not include changes to the main TDVT page I am planning. I originally had a FAQ here, but it was duplicating the FAQ on the main TDVT page, so I'm working on updating that one.

Pictures:
tdvt_test_details_worksheet
tdvt_usage_guide

### CODE REVIEW

File: docs/tdvt-test-case.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+## Case 3: The error originated from the driver or database, Generated SQL was generated and actual tuples is Null
+If an error happened in the driver or database, you'll need to us your knowledge of the database to triage the issue.
+
+One common database error happens when the dialect is incorrect and sends bad SQL commands that the database can't parse. This can happen a few different ways:
+- The mapping from Tableau function to your database's SQL protocol is incorrect. In this case, fixing the function in your dialect will fix the issue.
+- Tableau constructed the SQL statement wrong, or in a way your database does not support. If the generated SQL includes an incorrect SELECT statement or GROUPS BY something it shouldn't, you probably need to change your connector's capabilities. Search the [capabilities reference]({{ site.baseurl }}/docs/capabilities) list to find the relevant Tableau capability.
+
+## Case 4: The error originated from Tableau, no Generated SQL was generated and actual tuples is Null
+Other errors may occur. In most cases, the error message should be enough to diagnose and fix the issue. If not, you can open a github issue to ask us about the error.
```
Comment (lukewrites) [inline docs/tdvt-test-case.md:88]:
```suggestion
Other errors may occur. In most cases, the error message should be enough to diagnose and fix the issue. If not, you can [open a GitHub issue](https://github.com/tableau/connector-plugin-sdk/issues) to ask us about the error.
```

**Review by lukewrites — APPROVED** [review]
Great addition to the SDK, thanks for doing it!

File: docs/tdvt-test-case.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+Before digging too deep, consider the following cases:
+- Precision errors.
+A common problem is precision errors for floating point values, although the test framework imposes some rounding to standardize results.
+If the error is not significant then you can either skip the test or add a new expected file.
+
+- Misaligned date functions.
+Another common problem involves various date functions that might be off by a day. These can be caused by start of week errors (Sunday should be 1 and Saturday is 7). Other causes might include discrepancies around some start of week calculations or quarter boundaries. You may need to change your TDS file's connection properties to have Sunday be the first day of the week (instructions can be found [here](https://help.tableau.com/current/pro/desktop/en-us/date_properties.htm))
+
+If those cases do not apply, you'll have to look deeper at the results you got and why it isn't matching our expected values. Something that can help is to do the calculation on paper.
```
Comment (rosswbrown) [inline docs/tdvt-test-case.md:64]:
```suggestion
If those cases do not apply, you'll have to look deeper at the results you got and why it isn't matching the expected values. Something that can help is to do the calculation on paper.
```

**Review by rosswbrown — APPROVED** [review]

File: docs/tdvt.md [hunk 3]
```diff
@@ -176,132 +174,134 @@ This is a selection of the output:
...
+LogicalExclusions_Staples | Filter.Trademark | N/A | Exclude logical tests that target the Staples table | No
+ExpressionExclusions_Standard | string.ascii,string.char | N/A | Exclude expression tests | No
+
+### [LODTests]
+Add this section to run LOD tests. This section is required.
+
+Option | Example Value | Default | Description | Required?
+-|-|-|-|-
+LogicalExclusions_Staples | Filter.Trademark | N/A | Exclude logical tests that target the Staples table | No
+ExpressionExclusions_Calcs | string.ascii,string.char | N/A | Exclude expression tests that target the Calcs table| No
```
Comment (rosswbrown) [inline docs/tdvt.md:237]: I didn't see any examples of `ExpressionExclusions_Calcs` usage today

File: docs/tdvt.md [hunk 4]
```diff
@@ -176,132 +174,134 @@ This is a selection of the output:
...
-#You can put in comma-separated string values that match part or all of a test name to exclude them. The asterisk works as a wildcard.
-
-LogicalExclusions_Calcs = string.right
-
-LogicalExclusions_Staples = Filter.Trademark
-
-ExpressionExclusions_Standard = string.ascii,string.char,logical
-
-#If you remove this section then your test sute will not include the Level of Detail tests.
+ExpressionExclusions_Standard = string.isdate,date.datediff.day,date.datediff.dayofyear,date.datediff.hour,date.datediff.minute,date.datediff.second,date.datediff.sow.day,date.datediff.sow.dayofyear,date.datediff.weekday,date.datediff.sow.hour,date.datediff.sow.minute,date.datediff.sow.second,date.datediff.sow.weekday,datetime.cast.float
```
Comment (rosswbrown) [inline docs/tdvt.md:272]: Maybe shorten this example slightly or use the `*` notation

File: docs/tdvt.md [hunk 5]
```diff
@@ -451,69 +456,58 @@ You can re-map these in the TDS file by adding a column definition like this:
...
-__Troubleshoot test failures__  
-Check these things: 
-- Ensure that the data is loaded correctly.
-Incorrect data can cause many test failures.
-
-- Run and pass calcs_data.   
-The standard test suite include a test named calcs_data that contains several test cases, one for each column in the table.
+__Check that test data is loaded correctly__
+Incorrect data can cause many test failures. There are two tests that test if your data is loaded correctly:
+- calcs_data.
```
Comment (rosswbrown) [inline docs/tdvt.md:461]:
```suggestion
- calcs_data
```

File: docs/tdvt-test-case.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+**Note:** Some calculations return multiple values. In this case, Tableau merges all duplicate values together and then sorts them.
+
+**Note:** The expected values listed in Actual Tuples are pulled from the first expected file, which may not be the closest matching expected file. For things like dates that have a lot of potential formats, this could mean your actual results look nothing like the expected. In this case, use the "Closest Expected" column in the test case and look up the correct expected file.
+
+## Case 2: Error message is "No such function...", no Generated SQL was generated and actual tuples is Null
+This error occurs if your dialect does not implement a function and Tableau tries to use it. Adding the function to the dialect will fix this issue.
+
+## Case 3: The error originated from the driver or database, Generated SQL was generated and actual tuples is Null
+If an error happened in the driver or database, you'll need to us your knowledge of the database to triage the issue.
```
Comment (saugatt) [inline docs/tdvt-test-case.md:81]: There seems to be a typo of us instead of use here.

**Review by rosswbrown — APPROVED** [review]

File: docs/tdvt-test-case.md (hunk 6 above)
Comment (pvanderknyff) [reply docs/tdvt-test-case.md:81]: Fixing, thanks!

File: docs/tdvt.md (hunk 3 above)
Comment (pvanderknyff) [reply docs/tdvt.md:237]: I copy/pasted the keyword from the example, guessing it's from a really old version of TDVT if it isn't being used in favor of ExpressionExclusions_Standard. I'll change this.

File: docs/tdvt.md (hunk 4 above)
Comment (pvanderknyff) [reply docs/tdvt.md:272]: I'll take a few of them out, thanks

**Pull request merged**